app/views/courses/show.html.erb

```diff
@@ -0,0 +1,28 @@
+<h1><%= @course.display_name %></h1>
+<hr>
+<% if current_user.nil? %>
+<div>You need to sign in to be able to join this class.</div>
+<br>
+<div class="sign-in-panel">
+<% if Rails.env == "development" %>
+<%= link_to "Developer Login", "/home/developer_login" %><br />
+<% end %>
+<%= render "devise/shared/links" %>
```

Comment:
It's less that it's a lot of code and more that we're duplicating it.  For a while, we were loading assessments and submissions in different ways depending on the controller/action, which means you had inconsistencies (students could access unreleased assessments, for example).